src/mpid/ch4/netmod/ofi/ofi_spawn.h

```diff
@@ -153,30 +151,10 @@ static inline int MPIDI_OFI_dynproc_create_intercomm(const char *port_name,
         break;
     }
 
-    int avtid;
-    avtid = 0;
-    MPIDIU_new_avt(entries, &avtid);
-
-#ifdef MPIDI_OFI_CONFIG_USE_AV_TABLE
-    MPIDI_OFI_CALL(fi_av_insert(MPIDI_Global.av, addr_table, entries, NULL, 0ULL, NULL), avmap);
-#else
-    fi_addr_t *mapped_table;
-    mapped_table = (fi_addr_t *) MPL_malloc(entries * sizeof(fi_addr_t));
-
-    MPIDI_OFI_CALL(fi_av_insert(MPIDI_Global.av, addr_table, entries,
-                                mapped_table, 0ULL, NULL), avmap);
-    for (i = 0; i < entries; i++) {
-        MPIDI_OFI_AV(&MPIDIU_get_av(avtid, i)).dest = mapped_table[i];
-    }
-    MPL_free(mapped_table);
-#endif
-
-    MPIDIU_update_node_map(avtid, entries, node_table);
-
```

Comment:
`MPIDIU_upids_to_lupids()` in `ch4/src/ch4r_comm.h` is what you are looking for. It internally call the `MPIDI_NM_upids_to_lupids()` function so that the netmod can add the received address to the address vector. After the netmod part, it also takes the responsibility of updating the node map.

Terms used in the patch (these are introduced in the previous LUPID patch).
1. UPID: It essentially is the network address, which is the entry that being inserted into the address vector. The array of UPIDs in the code is the `local_upids` and `remote_upids`. Because UPIDs can have variable length, there are also `local_upid_size` array and `remote_upid_size` array.
2. LUPID: It is a packed form of <avtid,lpid> pair.

The general idea is that the upper layer should only need to use LUPID. For example, in `MPI_Group` and `MPI_Intercomm_create`. UPID is only used within CH4.

The old `ofi_spawn.h` does the following steps in connect/accept:
* address exchange between local and remote roots (use OFI routines directly)
* root broadcast the received addresses to local group (a couple MPIR_Bcast_intra calls)
* insert addresses to AV and create map the intercomm

The new code keep the workflow while trying to reuse some CH4 functions for intercomm address exchange:
* UPID exchange between root (mostly the same with old version, but with the support of variable UPID length); insert address using `MPIDIU_upids_to_lupids` (shared CH4 function)
* broadcast UPIDs in local group (reusing `MPIDIU_Intercomm_map_bcast_intra` which internally calls `MPIDIU_upids_to_lupids`) <- I feel this internal call is nasty when I try to explain this. Probably should separate it out.
* create intercomm using LUPIDs (obtained from `MPIDIU_upids_to_lupids`)

Beside this, I have also added a barrier at the end of accept/connect. Without this, the root rank will always go to SHM init and blocked on `PMI_Barrier` while the other ranks waiting the broadcast to progress.